public/app/features/correlations/Forms/ConfigureCorrelationSourceForm.tsx

```diff
@@ -71,34 +83,43 @@ export const ConfigureCorrelationSourceForm = () => {
         />
 
         <Field
-          label="Results field"
-          description="The link will be shown next to the value of this field"
+          label={t('correlations.source-form.results-label', 'Results field')}
+          description={t(
+            'correlations.source-form.results-description',
+            'The link will be shown next to the value of this field'
+          )}
           className={styles.label}
           invalid={!!formState.errors?.config?.field}
           error={formState.errors?.config?.field?.message}
         >
           <Input
             id={getInputId('field', correlation)}
-            {...register('config.field', { required: 'This field is required.' })}
+            {...register('config.field', {
+              required: t('correlations.source-form.results-required', 'This field is required.'),
+            })}
             readOnly={readOnly}
           />
         </Field>
         {variables.length > 0 && (
           <Card>
-            <Card.Heading>Variables used in the target query</Card.Heading>
+            <Card.Heading>
+              <Trans i18nKey="correlations.source-form.heading">Variables used in the target query</Trans>
+            </Card.Heading>
             <Card.Description>
-              You have used following variables in the target query:{' '}
-              {variables.map((name, i) => (
-                <span className={styles.variable} key={i}>
-                  {name}
-                  {i < variables.length - 1 ? ', ' : ''}
-                </span>
-              ))}
-              <br />A data point needs to provide values to all variables as fields or as transformations output to make
-              the correlation button appear in the visualization.
-              <br />
-              Note: Not every variable needs to be explicitly defined below. A transformation such as{' '}
-              <span className={styles.variable}>logfmt</span> will create variables for every key/value pair.
+              <Trans i18nKey="correlations.source-form.description">
+                You have used following variables in the target query:{' '}
+                {variables.map((name, i) => (
+                  <span className={styles.variable} key={i}>
+                    {name}
+                    {i < variables.length - 1 ? ', ' : ''}
+                  </span>
+                ))}
+                <br />A data point needs to provide values to all variables as fields or as transformations output to
+                make the correlation button appear in the visualization.
+                <br />
+                Note: Not every variable needs to be explicitly defined below. A transformation such as{' '}
+                <span className={styles.variable}>logfmt</span> will create variables for every key/value pair.
+              </Trans>
```

Comment:
Because of the `{variables.map(...)}` this does not work. Will need to research and get back to you on how this should be done instead.

Reply:
I think we can try to extract it to a separate variable.

For the record this is how it renders now:

<img width="1059" alt="Screenshot 2023-10-06 at 09 02 42" src="https://github.com/grafana/grafana/assets/745532/01c70e9a-e777-4545-8c90-636a76dd6a67">

Reply:
Sorry yes, I investigated but never got back. 

The JSX node will either have to be put into a seperate variable and can be interpolated into the phrase, something like

```jsx
const variablesList = variables.map(name => <span>...</span>)
...
return (
  ...
  <Trans i18nKey="correlations.source-form.description">
    You have used the following variables: {variablesList} ...
  </Trans>
)
```

I'm not 100% sure whether this would work or not. The other option which should definitely work is to split the list out into a seperate component:

```tsx
function VariableList({variables}) {
  return <>
    {variables.map(name => <span>...</span>)} etc
  </>
}

function ConfigureCorrelationSourceForm() {
  ..
  return (
    ...
    <Trans i18nKey="correlations.source-form.description">
      You have used the following variables: <VariableList variables={variables} /> ...
    </Trans>
)
}